test/unit-test/signaling_api/signaling_api_utest.c

```diff
@@ -75,3 +75,1168 @@ void test_signaling_ConstructDescribeSignalingChannelRequest_BadParams( void )
+ */
+void test_signaling_ParseDescribeSignalingChannelResponse_BadParams( void )
+{
+    const char message;
+    size_t messageLength = sizeof( message );
+    SignalingChannelInfo_t channelInfo;
+    SignalingResult_t result;
+
+    result = Signaling_ParseDescribeSignalingChannelResponse( NULL,
+                                                              messageLength, &( channelInfo ) );
+
+    TEST_ASSERT_EQUAL( SIGNALING_RESULT_BAD_PARAM,
+                       result );
+
+    result = Signaling_ParseDescribeSignalingChannelResponse( &( message ),
+                                                              messageLength, NULL );
+
+    TEST_ASSERT_EQUAL( SIGNALING_RESULT_BAD_PARAM,
+                       result );
+}
+
+/*-----------------------------------------------------------*/
+
+/**
+ * @brief Validate Signaling Parse Describe Response functionality for Invalid JSON.
+ */
+void test_signaling_ParseDescribeSignalingChannelResponse_InvalidJson( void )
+{
+    const char * message = "{\"ChannelInfo\": {";        /* Missing Closing Braces */
+    size_t messageLength = strlen( message );
+    SignalingChannelInfo_t channelInfo;
+    SignalingResult_t result;
+
+    result = Signaling_ParseDescribeSignalingChannelResponse( message,
+                                                              messageLength, &( channelInfo ) );
+
+    TEST_ASSERT_EQUAL( SIGNALING_RESULT_INVALID_JSON,
+                       result );
+
+    /* <--------------------------------------------------------------------> */
+
+    const char *message2 = "{"
+        "\"ChannelInfo\": {"
+            "\"SingleMasterConfiguration\": \"InvalidValue\""
+        "}"
+    "}";
+    size_t messageLength2 = strlen( message2 );
+
+    result = Signaling_ParseDescribeSignalingChannelResponse( message2,
+                                                              messageLength2, &( channelInfo ) );
+
+    TEST_ASSERT_EQUAL( SIGNALING_RESULT_INVALID_JSON,
+                       result );
+}
+
+/*-----------------------------------------------------------*/
+
+
+/**
+ * @brief Validate Signaling Parse Describe Response functionality for Unexpected Response.
+ */
+void test_signaling_ParseDescribeSignalingChannelResponse_UnexpectedResponse( void )
+{
+    const char * message = "{\"UnexpectedKey\": {}}";            /* The Key is not known */
+    size_t messageLength = strlen( message );
+    SignalingChannelInfo_t channelInfo;
+    SignalingResult_t result;
+
+    result = Signaling_ParseDescribeSignalingChannelResponse( message,
+                                                              messageLength, &( channelInfo ) );
+
+    TEST_ASSERT_EQUAL( SIGNALING_RESULT_UNEXPECTED_RESPONSE,
+                       result );
+}
+
+/*-----------------------------------------------------------*/
+
+/**
+ * @brief Validate Signaling Parse Describe Response functionality for Invalid TTL.
```

Comment:
Add a note here to describe valid TTL range.

Reply:
Done, will be reflected in the next PR. Thanks